### PR TITLE
Updated assignment of number of GBTs for different gemTypes

### PR DESCRIPTION
**Pull request by sbutalla**
## Description
Added a logic statement to correctly assign the number of GBTs for the appropriate gemType (i.e., ge21 has two GBTs and ge11 has three GBTs).

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context
Number of GBTs was being assigned incorrectly to the ge21 OH as three instead of two.

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: gempython/tools/amc_user_functions_xhal.py [hunk 1]
```diff
@@ -279,7 +274,15 @@ def getGBTLinkStatus(self,doReset=False,printSummary=False, ohMask=None):
...
-        NGBT = gbtsPerGemVariant["ge11"]
+        #Determine the number of GBTs based on the gemType
+        if gemType == "ge11":
+            NGBT = gbtsPerGemVariant["ge11"]
+        elif gemType == "ge21":
+            NGBT = gbtsPerGemVariant["ge21"]
+        else:
+            NGBT = 0
+            print("Unrecognized gemType, setting NGBT = 0")
+
```
Comment (bdorney) [inline gempython/tools/amc_user_functions_xhal.py]: Would strongly prefer the following, and in the `HwAMC::__init__(...)` method:

```python
if gemType in gbtsPerGemVariant.keys():
    self.NGBT = gbtsPerGemVariant[gemType]
else:
    raise KeyError("Unrecognized gemType {0}".format(gemType))
```

Then here you just use `self.NGBT`

**Review by bdorney — DISMISSED** [review]
changes should go in `init` method and raise an exception if unrecognized key rather than just continuing.

File: gempython/tools/amc_user_functions_xhal.py (hunk 1 above)
Comment (bdorney) [reply gempython/tools/amc_user_functions_xhal.py]: An alternative, but also acceptable implementation in `HwAMC::_init__(...), could also be:

```python
try:
   self.NGBT = gbtsPerGemVariant[gemType]
except KeyError as error:
   printRed("Unrecognized gemType {0}".format(gemType))
   exit(os.EX_DATAERR)
```

You might need to add `import os` and `from gempython.utils.gemLogger import printRed`.  I believe `printRed` is defined under `gempython/utils/gemLogger.py` but double check.

Comment (sbutalla) [reply gempython/tools/amc_user_functions_xhal.py]: I agree, I will implement and resubmit the pull request.

File: gempython/tools/amc_user_functions_xhal.py [hunk 2]
```diff
@@ -179,6 +179,12 @@ def __init__(self, cardName, debug=False):
         if debug:
             print "My FW release major = ", self.fwVersion
 
+        #Determine the number of GBTs based on the gemType
```
Comment (mexanick) [inline gempython/tools/amc_user_functions_xhal.py:182]: I don't see how the gemType is passed to the constructor of the `HwAMC` class. Please update the `init` signature

**Review by mexanick — CHANGES_REQUESTED** [review]

File: gempython/tools/amc_user_functions_xhal.py (hunk 2 above)
Comment (sbutalla) [reply gempython/tools/amc_user_functions_xhal.py:182]: Fixed! Thank you for spotting that.

**Review by mexanick — APPROVED** [review]

**Review by mexanick — CHANGES_REQUESTED** [review]
the gemType should be declared as local variable, not as `args.gemType`. 
I.e.: 
```python 
def __init__(self, cardName, debug=False, gemType="ge11"):
```

**Review by mexanick — APPROVED** [review]

**Review by mexanick — APPROVED** [review]

**Review by bdorney — COMMENTED** [review]
Either @sbutalla or a maintainer should `squash` the commit history to a single commit with a meaningful commit message.

**Comment by mexanick** [comment]
my bad, forgot to squash

**Pull request merged**